hedera-node/src/main/java/com/hedera/services/state/merkle/internals/CopyOnWriteIds.java

```diff
@@ -172,9 +172,9 @@ public void setNativeIds(long[] ids) {
 		return ids;
 	}
 
-	public List<TokenID> getAsIds() {
+	public List<TokenID> getAsIds(int count) {
 		final List<TokenID> modelIds = new ArrayList<>();
-		for (int i = 0, n = size(); i < n; i++) {
+		for (int i = 0, n = size(); i < count && i < n; i++) {
```

Comment:
`... n = Math.min(count, size()); i < n; ...`?

Reply:
Actually we no longer need this. Removing it